crates/fuel-core/src/schema/coins.rs

```diff
@@ -213,15 +214,23 @@ impl CoinQuery {
         #[graphql(desc = "\
             The list of requested assets` coins with asset ids, `target` amount the user wants \
             to reach, and the `max` number of coins in the selection. Several entries with the \
-            same asset id are not allowed.")]
-        query_per_asset: Vec<SpendQueryElementInput>,
+            same asset id are not allowed. The result can't contain more coins than `max_inputs`.")]
+        mut query_per_asset: Vec<SpendQueryElementInput>,
         #[graphql(desc = "The excluded coins from the selection.")] excluded_ids: Option<
             ExcludeInput,
         >,
     ) -> async_graphql::Result<Vec<Vec<CoinType>>> {
         let params = ctx
             .data_unchecked::<ConsensusProvider>()
             .latest_consensus_params();
+        let max_input = params.tx_params().max_inputs();
+
+        // `coins_to_spend` exists to help select inputs for the transactions.
+        // It doesn't make sense to allow the user to request more than the maximum number
+        // of inputs.
+        // TODO: To avoid breaking changes, we will truncate request for now.
+        //  In the future, we should return an error if the input is too large.
```

Comment:
Can we reference an issue here?

Reply:
https://github.com/FuelLabs/fuel-core/issues/2343

Reply:
Nice, thanks!